src/NSFW.cpp

```diff
@@ -310,11 +321,15 @@ void NSFW::pollForEvents() {
       }
     }
 
-    std::this_thread::sleep_for(std::chrono::milliseconds(sleepDuration));
+    mWaitPoolEvents.wait_for(lck,std::chrono::milliseconds(sleepDuration));
```

Comment:
Instead of `wait_for`, we probably want to use `wait_until` with a predicate. With the first, it's possible to have spurious wakeups (e.g. if `sleepDuration` is `1s`, we might have a spurious wakeup at `75ms` and return results early). With the second, we can detect if a spurious wakeup occurred and continue waiting.

Something like
```
const auto waitUntil = std::chrono::steady_clock::now() + sleepDuration;
mWaitPoolEvents.wait_until(lck, waitUntil, [this, waitUntil]{
  return !mRunning || std::chrono::steady_clock::now() >= waitUntil;
});
```

https://en.cppreference.com/w/cpp/thread/condition_variable/wait_for
https://en.cppreference.com/w/cpp/thread/condition_variable/wait_until

> It may also be unblocked spuriously. When unblocked, regardless of the reason, lock is reacquired and wait_for() exits

Additionally: we need to protect the setting of `mRunning` with the same mutex we're using in `lck`, even though it's atomic. Consider the case where we have two threads: main thread and loop thread. Loop thread pauses at the point where the predicate checks and sees that `mRunning == false` but hasn't gotten to the point where `wait_until` starts waiting again. At that point in main, we set `mRunning = true` and also call `mWaitPoolEvents.notify_one()`. After this, loop thread resumes and starts waiting on `mWaitPoolEvents` for a notify that will never come.

https://en.cppreference.com/w/cpp/thread/condition_variable
> Even if the shared variable is atomic, it must be modified under the mutex in order to correctly publish the modification to the waiting thread.

Reply:
Ok, Changed in commit: `Change wait_for to wait_until to fix possible spurious wakeups`